src/Controls/src/Core/Compatibility/Handlers/Shell/iOS/UIContainerView.cs

```diff
@@ -90,9 +90,27 @@ public override void WillMoveToSuperview(UIView newSuper)
 			ReMeasure();
 		}
 
+		bool _currentlyInLayoutSubviews;
+		public override void SetNeedsLayout()
+		{
+			// Certain children (LayoutView) will call `SetNeedsLayout`
+			// On the parent during the LayoutSubviews phase
+			// which leads to an infinite layout loop
+			if (!_currentlyInLayoutSubviews)
+				base.SetNeedsLayout();
+		}
+
 		public override void LayoutSubviews()
 		{
-			_view.LayoutToSize(Width ?? Frame.Width, Height ?? MeasuredHeight);
+			try
+			{
+				_currentlyInLayoutSubviews = true;
+				_view.LayoutToSize(Width ?? Frame.Width, Height ?? MeasuredHeight);
```

Comment:
Just a guess, but I think the problem is that LayoutToSize is doing something weird where it manually sets the Frame for the platform view and _then_ calls Arrange on the IView. I think if you remove that explicit Frame settting and just call Arrange, it won't get stuck in a loop.

Reply:
Alright, if I just rely on arrange things seem to all behave a bit better 👍